resources/local/config.toml

```diff
@@ -310,6 +310,11 @@ enable_mem_deduplication = true
 # For example, setting this value to 5 means that every 5th time something is put in the pool the cache is swept.
 mem_pool_prune_interval = 4096
 
+# Parallel storage request execution.
+#
+# Controls the maximum number of threads that are spawned to access storage in parallel.
+max_sync_tasks = 32
```

Comment:
32 is just a guess, I assume it can be tweaked using load testing. The `spawn_blocking` maximum number of threads is fairly high, and it also shared across all blocking tasks, not just storage IO ones.

The default limit for `spawn_blocking` according to the docs is 512, which seems large for IO. If we were to bring this down to something in the realm of 32, the contention by CPU-bound tasks could be noticeable.

Looking at the (maybe somewhat dated) [LMDB benchmarks](http://www.lmdb.tech/bench/ondisk/#sec12), assuming 2000 bytes (which seems to be a reasonable size) VM reads, it seems that the sweet spot is somewhere between 32 and 64, with returns diminishing fast after 8:

![LMDB 2000 bytes VM parallel reads](http://www.lmdb.tech/bench/ondisk/VM2000rscale.png)

Reply:
32 is just a guess, I assume it can be tweaked using load testing. The `spawn_blocking` maximum number of threads is fairly high, and it also shared across all blocking tasks, not just storage IO ones.

The default limit for `spawn_blocking` according to the docs is 512, which seems large for IO. If we were to bring this down to something in the realm of 32, the contention by CPU-bound tasks could be noticeable.

Looking at the (maybe somewhat dated) [LMDB benchmarks](http://www.lmdb.tech/bench/ondisk/#sec12), assuming 2000 bytes (which seems to be a reasonable size) VM reads, it seems that the sweet spot is somewhere between 32 and 64, with returns diminishing fast after 8:

![LMDB 2000 bytes VM parallel reads](http://www.lmdb.tech/bench/ondisk/VM2000rscale.png)

Reply:
I'm fine with the separate semaphore, as it does provide finer-grained control over back pressure here.